scripts/api-compatibility-check/README.md

```diff
@@ -0,0 +1,83 @@
+# API Compatibility check
+
+## What is it
+
+API compatibility checks help us to ensure that we would not accidentally break a public API. The set of scripts runs Apple's tool `swift-api-digester` to dump public API to a reusable JSON file and compare each other.
+
+## How to run
+
+You can find two scripts in this folder. The first, `breaking-api-check.sh`, designed to run set of public API checks over the pre-built baseline. The second, `rebuild-baseline.sh`, helps to generate new baseline based on git reference. Let's take a closer look on each of them.
+
+### `breaking-api-check.sh`
+
+The `breaking-api-check.sh` accept optional argument `-p path` which should be packaged version of MapboxMaps. Without `-p` argument, script would trigger `scripts/release/packager/package-mapbox-maps.sh` to build XCFramework from scratch. It is useful to path pre-built binary to avoid extra building steps. Breaking API checks running against `.baseline.zip` JSON API digester dump. To update the baseline, you have to proceed to `rebuild-baseline.sh`.
+The script is path-independent so you can call it from any other folder with an absolute or relative path.
+
+### `rebuild-baseline.sh`
+
+The `rebuild-baseline.sh` script simplifies baseline data updates. Usually, we should not update the baseline for the major version. However, my investigations show that digester would trigger breaking changes for `@_spi` changes as well which is not the desired behavior.
+To update the baseline, update the `.baseapi` file to include a new git reference, then call the script. The new git worktree would be checkout in a hidden folder for script purposes. 
+In a successful result, you would get an updated version of `.baseline.zip`. This archive represents folder structure with JSON SDK API dump made by swift-api-digester. It is a native structure for `swift-api-digester` so we can easily extend our platform validations to include macosx/watchos or any other Apple platform.
+
+## What is `swift-api-digester`
+
+Swift API digester is an official tool to dump public API to JSON representation based on AST and compare dumps if needed.
+
+### Limitations
+
+`swift-api-digester` tool lives in the opensource swift toolchain for quite a long but wasn't included in Xcode distribution until Xcode 13. That's why you have to use Xcode 13 or newer to run breakage checks.
+Another limitation is that you must use `swift-api-digester` from the same toolchain you built your SDK. You cannot run Xcode 13 digester over the SDK built with Xcode 12.5.1.
+
+### Where to find `swift-api-digester`
+
+To get access to the `swift-api-digester` you have to make a path manually. The handy shortcut can be:
+
+```bash
+API_DIGESTER_PATH="$(xcode-select -p)/Toolchains/XcodeDefault.xctoolchain/usr/bin/swift-api-digester"
+```
+
+There is no access to `switf-api-digester` through the `$PATH` or even `xcrun`.
```

Comment:
```suggestion
There is no access to `swift-api-digester` through the `$PATH` or even `xcrun`.
```

Reply:
`swift package experimental-api-diff` is a new tool to run `swift-api-digester` in a simple way. You just passing a baseline tag to the `api-diff` and it would build both dumps and compare them.

Reply:
Should we use that instead?